crates/wasm-smith/src/lib.rs

```diff
@@ -814,16 +819,33 @@ where
     ) -> Result<Rc<ModuleType>> {
         let exports = self.arbitrary_instance_type(u, entities)?;
         let mut imports = Vec::new();
+        let mut import_types = indexmap::IndexMap::new();
         let mut names = HashMap::new();
         if !entities.max_reached(&self.config) {
             arbitrary_loop(u, 0, self.config.max_imports(), |u| {
                 let (module, name) = unique_import_strings(1_000, &mut names, true, u)?;
                 let ty = self.arbitrary_entity_type(u, entities)?;
+                if let Some(name) = &name {
+                    let ity = import_types.entry(module.clone()).or_insert_with(|| {
+                        EntityType::Instance(u32::max_value(), Default::default())
+                    });
```

Comment:
In general there's a fair amount of cloning happening everywhere in wasm-smith wrt the module structure and module linking, so I'm not too worried about this specifically. The default limits in place and `SwarmConfig` I think makes it so we don't have to worry about this too too much. That being said though I'm mostly relying on oss-fuzz's timeouts to tell us where we need to optimize here, because there's definitely more than one place we can optimize in this file!

For now though I think I'll leave this as-is because I think it's the most readable, but if it becomes problematic I'll probably refactor this out so it's not all in one big blob.